db/version_edit.cc

```diff
@@ -97,6 +105,23 @@ bool VersionEdit::EncodeTo(std::string* dst) const {
   if (has_max_column_family_) {
     PutVarint32Varint32(dst, kMaxColumnFamily, max_column_family_);
   }
+  if (has_deleted_log_number_) {
+    // TODO(myabandeh): uncomment me when manifest is forward-compatible
+    // PutVarint32Varint64(dst, kDeletedLogNumber, deleted_log_number_);
+    // Since currently manifest is not forward compatible we encode this entry
+    // disguised as a kNewFile4 entry which has forward-compatible extensions.
+    PutVarint32(dst, kNewFile4);
+    PutVarint32Varint64(dst, 0u, 0ull); // level and number
+    PutVarint64(dst, 0ull); // file size
+    InternalKey dummy_key(Slice("dummy_key"), 0ull, ValueType::kTypeValue);
+    PutLengthPrefixedSlice(dst, dummy_key.Encode()); // smallest
+    PutLengthPrefixedSlice(dst, dummy_key.Encode()); // largest
+    PutVarint64Varint64(dst, 0ull, 0ull); // smallest_seqno and largerst
+    PutVarint32(dst, CustomTag::kDeletedLogNumberHack);
+    PutLengthPrefixedSlice(dst, Slice()); // custom field slice
```

Comment:
I think this length-prefixed slice is supposed to be the whole contents of your custom record type. Otherwise earlier versions of the decoding code can't know where your custom record ends.